paasta_tools/cli/utils.py

```diff
@@ -1083,3 +1101,31 @@ def get_paasta_oapi_api_clustername(cluster: str, is_eks: bool) -> str:
     "eks-" prefix
     """
     return f"eks-{cluster}" if is_eks else cluster
+
+
+def get_current_ecosystem() -> str:
+    """Infer ecosystem from local-run configuration"""
+    local_run_config = load_system_paasta_config().get_local_run_config()
+    ecosystem = local_run_config["default_cluster"].split("-", 1)[-1]
+    return f"{ecosystem}prod" if ecosystem == "corp" else ecosystem
+
+
+def get_service_auth_token() -> str:
+    """Uses instance profile to authenticate with Vault and generate token for service authentication"""
+    ecosystem = get_current_ecosystem()
+    vault_client = get_vault_client(get_vault_url(ecosystem), get_vault_ca(ecosystem))
+    vault_role = load_system_paasta_config().get_service_auth_vault_role()
+    metadata_provider = InstanceMetadataProvider(
+        iam_role_fetcher=InstanceMetadataFetcher(),
+    )
+    instance_credentials = metadata_provider.load().get_frozen_credentials()
+    vault_client.auth.aws.iam_login(
+        instance_credentials.access_key,
+        instance_credentials.secret_key,
+        instance_credentials.token,
+        mount_point="aws-iam",
```

Comment:
No action required just thinking out loud, this _could_ in theory change and be a different mount point like `aws`, especially if we wanted to switch to ec2 auth vs iam auth (in which case we'd also use `vault_client.auth.aws.ec2_login` instead of `iam_login`). This is re: [this comment](https://docs.google.com/document/d/1uOz83HMbj2jhDv8FB5oHjyXsg_LGyLv8COJHuwOrEPI/edit?disco=AAABNIwPENA) in the CEP but honestly i think there are more downsides than upsides to using ec2 auth so what we have here looks good to me, and if we decide to switch we will deal with that then :)